### PR TITLE
Guest music list

### LINKED ISSUE #237: Music list is not updated when one of many devices disconnects.
After device disconnect music list shows no color next to the song but song should not display. 

should_be_missing_songs

### PR DESCRIPTION
**Pull request by reidbaker**
Fixes #237

Basically what is happening is when we update the playlist it is an update not a replace so the removed songs don't get removed. 
The way we fixed this was when we get a updated user list message we calculate the users that were removed and include them in the userlist updated intent. In music library service we loop through the removed users and remove any songs that are their's.

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/model/User.java [hunk 1]
```diff
@@ -55,5 +55,17 @@ public String getMacAddress(){
...
-    }  
+    }
+
+    @Override
+    public boolean equals(Object obj){
+        boolean result;
+        if (obj == null) {
+            result = false;
+        } else {
+            User other = (User) obj;
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/model/User.java:66]: Needs a test for if obj instanceof User...if it isnt, return false.

File: SoundStream/src/com/lastcrusade/soundstream/service/MusicLibraryService.java [hunk 2]
```diff
@@ -157,11 +158,24 @@ public void onReceiveAction(Context context, Intent intent) {
                     }
                 }
             })
+            .addLocalAction(UserList.ACTION_USER_LIST_UPDATE, new IBroadcastActionHandler() {
+
+                @Override
+                public void onReceiveAction(Context context, Intent intent) {
+                    UserList removedUsers = (UserList) intent.getParcelableExtra(UserList.EXTRA_REMOVED_USERS);
+                    if(removedUsers != null) {
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/service/MusicLibraryService.java:170]: Can you add a comment here stating what this is doing?

File: grail-diary [hunk 3]
```diff
@@ -0,0 +1 @@
+Subproject commit 99ed48b2a364d99239122b54bfe63217e712298b
```
Comment (theJenix) [inline grail-diary]: What is this?

**Comment by theJenix** [comment]
I saw one potential big issue in the User equals method, and would like a comment in the MusicLibraryService where we process the removed list.  It also looks like theres an extraneous file (grails diary?)  Otherwise, I dig :it: and it's :ok_hand: with me.

**Comment by dgreenhalgh** [comment]
Great emoji use here, @theJenix :hamburger:

**Comment by theJenix** [comment]
I'm a profejional.

File: grail-diary (hunk 3 above)
Comment (reidbaker) [inline grail-diary]: Dont know how that got in there. I am trying to figure out how to remove it.

**Comment by reidbaker** [comment]
:ok_woman: think that is all the updates people asked for

**Comment by theJenix** [comment]
Fine like :wine_glass: :thumbsup: :ok_hand: :ship:s ahoy.

**Comment by dgreenhalgh** [comment]
:goat: :goat: :goat: :goat: :goat: :goat: :goat: :goat: :goat: :goat: :goat:

**Pull request merged**